chimney/src/main/scala/io/scalaland/chimney/dsl/TransformerDefinition.scala

```diff
@@ -18,6 +18,18 @@ final class TransformerDefinition[From, To, C <: TransformerCfg](
     val instances: Map[(String, String), Any]
 ) {
 
+  def __refineConfig[C1 <: TransformerCfg]: TransformerDefinition[From, To, C1] =
```

Comment:
Add comment that these are public only because macro requires it to be public.

Actually, we could probably rewrite it to be extension method of some sort but stored in companion object, so that it would be always available and had access to private fields (and name these extension methods something like "InternalSth")

Reply:
Such extension methods would be still visible in IDE code completion, so benefit would be minimal (if any).

But the trick with hiding these methods in some implicit class (which is not automatically imported) could be possible. The macro will need to ensure that it's imported, when needed; it requires prefix inspection (to avoid importing it several times), but this way we could hide these methods from appearing in IDE.

I will give it a try later.

Reply:
In fact, implicit part is not absolutely crucial to this idea. We can have extension class (`TransformerDefinitionOps`) without implicits, as we drive the code generation here. Instead of `td.__refineConfig[C]` we just emit `TransformerDefinitionOps(td).refineConfig[C]`.

This way, we still don't need to put `F` type tree to emitted code, while remove these internal `__` methods from public API.

Reply:
I've tried this. It works well in 2.12/2.13, but due to some problems with type constructors inference, it fails in 2.11. It applies both to first idea (with implicit class + import) and to second (plain class + new in generated code).

As for now, I think that keeping these methods in public api (until we drop 2.11) is lesser evil. I will add proper comments though.